### PR TITLE
Fix orpha phenomodels

### LINKED ISSUE #2159: Use Orphanet to describe diseases, together with OMIM
To do:
- Download all Orphanet diseases and add them to the database (different collection from the OMIM one?)

- HPO terms to Orphanet diseases data is included in the genes_to_phenotype.txt file already used to couple genes to OMIM diseases:

image

- Make terms available at the case level to assign diseases

### PR DESCRIPTION
**Pull request by TereseBo**
This PR adds the possibility to include orpha disease checkboxes to phenomodels. A radio button for selecting orpha_terms have been added to the phenomodels page. The form and functions previously in place for omim checkboxes have been renamed omim-> diseases and adapted to handle both orpha and omim terms.

This PR closes  #2159

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Navigate to phenomodels page and verify hop and disease checkboxes can be added and removed from the phenomodels.

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by TB

### CODE REVIEW

**Comment by TereseBo** [comment]
On the main branch hpo and omim checkboxes can be added:
image


image

On this branch, ORPHA checkbox option is available:
image

The form to add a disease checkbox updates the placeholder and button to match the selected disease coding system when the type of disease checkbox is switched:
image

Hpo, omim and orpha terms can be added to the sub panel:
image

The checkboxes are saved with the correct type in DB:
image

File: scout/server/blueprints/phenomodels/controllers.py [hunk 1]
```diff
@@ -122,7 +119,7 @@ def _update_subpanel(subpanel_obj, supb_changes):
         all_terms = {}
         # loop over the terms to keep into the checboxes dict
         for child in children_list:
-            if child.startswith("OMIM"):
+            if child.startswith("OMIM" or "ORPHA"):
```
Comment (northwestwitch) [inline scout/server/blueprints/phenomodels/controllers.py:122]:
```suggestion
            if child.startswith("OMIM") or child.startswith("ORPHA"):
```

You can try also testing locally. The code you wrote causes the error I described as a comment to this PR. Tried with the code in my suggestion and it works!

**Review by northwestwitch — CHANGES_REQUESTED** [review]
Nice addition to the phenomodels feature! 🥇 

There is an error in the code that causes ORPHA checkboxes to never be saved from the phenomodels preview. I've suggested the fix to that. Otherwise looks great!

File: scout/server/blueprints/phenomodels/controllers.py (hunk 1 above)
Comment (TereseBo) [reply scout/server/blueprints/phenomodels/controllers.py]: Fixed according to suggestion

File: scout/server/blueprints/phenomodels/controllers.py [hunk 2]
```diff
@@ -131,7 +128,7 @@ def _update_subpanel(subpanel_obj, supb_changes):
             try:
                 node = Node(child, parent=root, description=term_obj["description"])
             except Exception:
-                flash(f"Term {child} could not be find in database")
+                flash(f"Term {child} could not be found in database")
```
Comment (northwestwitch) [inline scout/server/blueprints/phenomodels/controllers.py:131]: 💯

**Pull request merged**